src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1727,6 +1727,38 @@ export abstract class AbstractConfigureRecommendedExtensionsAction extends Actio
 			}));
 	}
 
+	protected getFolderRecommendedExtensions(extensionsFileResource: URI): TPromise<string[]> {
+		return this.fileService.resolveContent(extensionsFileResource)
+			.then(content => {
+				const folderRecommendations = (<IExtensionsContent>json.parse(content.value));
+				return folderRecommendations.recommendations || [];
+			}, err => []);
+	}
+
+	protected addRecommendedExtensionToFolder(extensionsFileResource: URI, extensionId: string): TPromise<any> {
+		return this.getOrCreateExtensionsFile(extensionsFileResource)
+			.then(({ created, content }) => {
+				const folderRecommendations: string[] = (<IExtensionsContent>json.parse(content)).recommendations || [];
+
+				if (folderRecommendations.indexOf(extensionId) !== -1) {
+					return TPromise.as(null);
+				}
+
+				// TODO:
+				// This will actually overwrite the contents of this key in the file,
+				// removing comments and any additional user modifications.
+				return this.jsonEditingService.write(extensionsFileResource,
```

Comment:
Will check on that

Reply:
I followed up on this in this comment and the case is not as bad as I initially thought it was: https://github.com/Microsoft/vscode/pull/50419#issuecomment-398504027.

In summary, the file is created if it doesn't exist, in that case there's nothing to overwrite so nothing is "lost". The created file wil have the instructions in comments since those are coming from the template, and will have the added recommendation too.

In the case of an existing files, the instructions or any comments that live outside of the `recommendations` key are not removed either because we are only replacing the contents of that key in the file and nothing else. In this image you can see the diff of the file when this action is executed. Notice how outside comemnts are preserved:

![image](https://user-images.githubusercontent.com/7514993/41617297-1bf40b40-73ce-11e8-8ac0-d71ed8ee7e64.png)

This TODO is referring to comments inside of the key. Not sure if we should aim to preserve those but wanted to bring it up anyway in case the team decided we should act on it or leave it as is.